feat: update run docs with --target options (#43)
terramate-io/terramate-docs: cli/reference/cmdline/run.md

  Use Terragrunt to generate the Terraform/OpenTofu Plan file.

### TMC: Deployment Targets

> **Note:** This is an upcoming experimental feature that is subject to change in the future. To use it now, you must enable the project config option `terramate.config.experiments = ["targets"]`

Reviewer: ```suggestion
**Note:** This is an experimental feature and may change. Enable it by setting the project config option `terramate.config.experiments = ["targets"]`.
```

Author: FYI I copy-pasted this from the scripts section, where it uses similar wording. But I use your suggestion anyway.

> Terramate supports synchronizing stacks to different deployment targets, each having an isolated state.
> This enables deploying to multiple environments, i.e. `testing`, `staging`, `prod`, or to multiple regions.

Reviewer: ```suggestion
Terramate supports synchronizing stacks to different deployment targets, each with an isolated state. It allows deployment to multiple environments, such as `testing`, `staging`, and `prod`, or to different regions.
```

> Before using this feature, you must enable targets for your project by setting the config option `terramate.config.cloud.targets.enabled = true`.

Reviewer: ```suggestion
Before using this feature, enable targets for your project by setting the config option `terramate.config.cloud.targets.enabled = true`.
```

Once enabled, the following options are available:

- `--target <target-identifier>`

>   Synchronize data to the Terramate Cloud (TMC) for the given deployment target. Stacks for each target are isolated from each other and have a separate status.

Reviewer: ```suggestion
  Synchronize data to the Terramate Cloud (TMC) for the given deployment target. Stacks for each target are isolated and have a separate status.
```

>   Once the targets feature is enabled, setting a target is _required_ for any option that uses Terramate Cloud, to safeguard against accidently synchronizing to an implicit default target.

Reviewer: ```suggestion
  After enabling the targets feature, you must specify a target for any option which uses Terramate Cloud to prevent accidental synchronization to the default target.
```

  Target identifiers must match the pattern `[a-zA-Z0-9_-]{1,64}`.

>   This flag is supported in combination with `--sync-deployment`, `--sync-drift-status`, `--sync-preview`, and `--status`.

Reviewer: ```suggestion
  `--target <target-identifier>` flag can be used with `--sync-deployment`, `--sync-drift-status`, `--sync-preview`, and `--status`.
```

Author: Using the wording thats used for other sections in this file to keep it consistent, but you could reword it everywhere in another PR.

- `--from-target <target-identifier>`

  Attempt to migrate stacks from `--from-target <old-target>` to `--target <new-target>`.
  

>   The rules for this migration are applied per stack and are defined as follows:
>   - If the stack already exists in `new-target`, the migration is skipped.
>   - If the stack doesn't exist in `old-target`, the migration is skipped.
>   - If used together with `--sync-preview`, the migration is skipped.
>   - Otherwise, the stack is moved/renamed from `old-target` to `new-target`.

Reviewer: ```suggestion
  The rules for this migration are applied per stack as follows:
- Skip the migration if the stack already exists in `new-target`.
- Skip the migration if the stack doesn't exist in `old-target`.
- Skip the migration if used with `--sync-preview`.
- Otherwise, move or rename the stack from `old-target` to `new-target`.
```

Author: Left the moved/renamed, its meant as a synonym rather than alternatives.

>   After the migration, any further operations are applied normally to `new-target`.

Reviewer: ```suggestion
  After the migration, all subsequent operations are applied to the `new-target` as usual.
```

>   This flag is only supported in combination with `--target`.

Reviewer: ```suggestion
 The `--from-target <target-identifier>` flag is supported only when used with `--target`.
```

Author: See above, leaving old wording for consistency.
